x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/ElementType.java

```diff
@@ -24,6 +24,8 @@ public enum ElementType {
     NULL("Null", (blockFactory, estimatedSize) -> new ConstantNullBlock.Builder(blockFactory)),
 
     BYTES_REF("BytesRef", BlockFactory::newBytesRefBlockBuilder),
+    CATEGORY_RAW("CategoryRaw", BlockFactory::newBytesRefBlockBuilder),
+    CATEGORY_INTERMEDIATE("CategoryIntermediate", BlockFactory::newBytesRefBlockBuilder),
```

Comment:
We talked in the past about associating an encoder/decoder to the byterefs blocks to encapsulate the specific content without introducing new types.
Would this work here?

Reply:
I *think* we can get away without introducing a new type by being more explicit when we're in the building code and building the `BlockHash` implementation "by hand" rather than going through the type generic methods. After all, this method is super special and those generic methods are built to handle the data regardless of it's source.

Reply:
To wrap this up: The current iteration doesn't put the information into the bytesref blocks - instead, we just don't extract the `CATEGORIZE` function into an eval and leave it inside the `Aggregate` logical plan. Later down the road, when mapping to physical operations, we detect the `CATEGORIZE` in the `AggregateExec` and use a custom block hash for `CATEGORIZE.